pkg/util/pki/csr.go

```diff
@@ -394,3 +389,24 @@ func SignatureAlgorithm(crt *v1alpha1.Certificate) (x509.PublicKeyAlgorithm, x50
 	}
 	return pubKeyAlgo, sigAlgo, nil
 }
+
+func GenerateSelfSignedCertFromCR(cr *v1alpha1.CertificateRequest, key crypto.Signer,
```

Comment:
This function is only used from a test package introduced in this PR. Given we already have SignCertificate etc., I think it may be better to remove this just to save adding more and more slight variations of the same functions to this package? It seems to be ever-growing, and we don't have a good consistency as to the types we do/don't accept 😬